src/common/dependency-manager/local-manager.ts

```diff
@@ -119,72 +82,69 @@ export default class LocalDependencyManager extends DependencyManager {
+    let service_node;
+    if (debug_path) {
+      const svc_path = path.join(path.dirname(this.config_path), debug_path);
+      service_node = await this.loadLocalService(svc_path);
+    } else if (this.linked_services.hasOwnProperty(service_name)) {
+      console.log(`Using locally linked ${chalk.blue(service_name)} found at ${chalk.blue(this.linked_services[service_name])}`);
+      service_node = await this.loadLocalService(this.linked_services[service_name]);
+    } else {
+      const [account_name, svc_name] = service_name.split('/');
+      const { data: service_digest } = await this.api.get(`/accounts/${account_name}/services/${svc_name}/versions/${service_tag}`);
+
+      const service_config = ServiceConfigBuilder.buildFromJSON(service_digest.config);
+      service_node = new ServiceNode({
+        service_config: service_config,
+        node_config: this.getNodeConfig(service_config),
+        tag: service_digest.tag,
+        image: service_digest.service.url.replace(/(^\w+:|^)\/\//, ''),
+        digest: service_digest.digest,
+      });
+    }
 
-  /**
-   * @override
-   */
-  async loadService(service_name: string, service_tag: string): Promise<ServiceNode> {
-    const [account_name, svc_name] = service_name.split('/');
-    const { data: service_digest } = await this.api.get(`/accounts/${account_name}/services/${svc_name}/versions/${service_tag}`);
-
-    const config = ServiceConfigBuilder.buildFromJSON(service_digest.config);
-    const node = new ServiceNode({
-      service_config: config,
-      tag: service_digest.tag,
-      image: service_digest.service.url.replace(/(^\w+:|^)\/\//, ''),
-      ports: {
-        target: config.getPort() || 8080,
-        expose: await this.getServicePort(),
-      },
-      parameters: await this.getParamValues(
-        `${config.getName()}:${service_digest.tag}`,
-        config.getParameters(),
-      ),
-    });
-    this.graph.addNode(node);
-    return node;
+    this.graph.addNode(service_node);
+    await this.loadDatastores(service_node);
+    if (recursive) {
+      await this.loadDependencies(service_node, recursive);
+    }
+    return service_node;
   }
 
-  protected loadParameters() {
+  protected async loadParameters() {
     for (const node of this.graph.nodes) {
       for (const [key, value] of Object.entries(node.parameters)) {
         // Only include in cli since it will read files off disk
         node.parameters[key] = readIfFile(value);
       }
     }
-    super.loadParameters();
+    await super.loadParameters();
+  }
+
+  /**
+  * @override
+  */
+  async loadServiceConfig(node_ref: string) {
+    const debug_path = this._environment.getServices()[node_ref]?.getDebugOptions()?.path;
+    if (debug_path) {
+      return ServiceConfigBuilder.buildFromPath(path.join(path.dirname(this.config_path), debug_path));
+    } else {
+      const [account_name, service_name, service_tag] = node_ref.split(/\/|:/);
```

Comment:
Same as above comment. I'd love to centralize this. I know it's not a lot of logic but it's highly specific and a regex that is not easy to read. Same with the `.replace(...)` methods for normalizing the ref, etc. We do string manipulation of the `service`, the `service:tag`, and the `account/service:tag` in many places.

Perhaps a utility class somewhere in the dep-manager with a handful of static methods. We have a lot of these across the codebase (CLI/API/UI). ie `ServiceNames.split(ref: string);`